lib/rules/yoda.js

```diff
@@ -20,6 +20,15 @@ function isComparisonOperator(operator) {
 }
 
 /**
+ * Determines whether an operator is an equality operator.
+ * @param {String} operator The operator to check.
+ * @returns {boolean} Whether or not it is an equality operator.
+ */
+function isEqualityOperator(operator) {
+    return (/^(==|===)$/).test(operator);
```

Comment:
I admit there's no consistentcy in the software world, but I don't consider `!=` an "equality operator". It's an "inequality operator".
More importantly, the use case for the option is to split off the operator that can be confused with the assignment operator, and that's `==`, not `!=`.
We can rename the option, if you wish.